File: apps/web/src/types/site.ts
import type { Template } from "./template";

export type SiteStatus = "draft" | "published" | "archived";

export interface Site {
  id: string;
  name: string;
  slug: string;
  description?: string;
  templateId: string;
  template?: Template;
  config: Record<string, unknown>;
  status: SiteStatus;
  domain?: string;
  publishedAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateSiteData {
  name: string;
  slug: string;
  description?: string;
  templateId: string;
  config?: Record<string, unknown>;
}

export interface UpdateSiteData {
  name?: string;
  description?: string;
  config?: Record<string, unknown>;
  status?: SiteStatus;
}

File: apps/server/src/main/java/com/xiao9/wb/site/entity/Site.java
package com.xiao9.wb.site.entity;

import com.xiao9.wb.template.entity.Template;
import com.xiao9.wb.user.entity.User;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "sites")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Site {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(length = 500)
    private String description;

    @Column(unique = true, length = 100)
    private String domain;

    @Column(unique = true, length = 100)
    private String subdomain;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private Status status = Status.DRAFT;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private SiteType siteType = SiteType.BLOG;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "template_id")
    private Template template;

    @CreatedDate
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;

    public enum Status {
        DRAFT,
        BUILDING,
        PUBLISHED,
        OFFLINE,
        ERROR
    }

    public enum SiteType {
        BLOG,
        CORPORATE,
        PORTFOLIO,
        ECOMMERCE
    }
}
